2016/proposals/rai-vivek.md

```diff
@@ -0,0 +1,275 @@
+# Result aggregation server for installation-scripts
+
+## Abstract
+
+Software is indispensable for any modern scientist irrespective of their field
+of work. However, many scientists lack or have limited exposure to tools and
+practices that would allow them to write more reliable and maintainable code
+with less effort.
+
+Software Carpentry is a collective effort of volunteer instructors, computer
+scientists and technology-enthusiasts with an aim of advancing research in
+scientific disciplines by teaching necessary computing skills.
+
+Software Carpentry works in this direction by organizing workshops, lessons or
+simply through discussions. 
+
+An important part of the workshop is to ensure that participants have installed
+all the necessary software. Software Carpentry has
+[installation-test](https://github.com/wking/swc-setup-installation-test)
+scripts that enables one to check if they have successfully installed the
+software required by their workshop. As an improvement to the these scripts, the
+proposal lays out following objectives:
+
+* Implement a server to store the results of installation-testing scripts (i.e.
+platform people use, which software dependency worked and which did not, error
+logs)
+* Enhance the installation-test scripts by adding an option for users to collect
+ results and send them to a server.
+
+## Technical Details
+
+The technical details of the project can be organized into following:
+
+* Collectable information - machine information, report
+* Backend - design of RESTful API, framework, database, schema
```

Comment:
Schema is quite important and I'd like you to think it through and write down somewhere here.

We'll certainly iterate through some schema changes, so it's important to decide if you're going to use SQLAlchemy + Alembic or not.

I don't want you to go into details like `title CHAR(53)` but lay down general relations (if required), decide what should be stored and how (e.g. store each field from `platform.uname` separately or concatenate it into one string and use separate field for OS class [MacOSX / Win / Linux]).